internal/operator/statefulset/builder.go

```diff
@@ -0,0 +1,871 @@
+		commandArgs = append(commandArgs, fmt.Sprintf("--etcd-client-cert-path=%s/tls.crt", common.EtcdClientTLSVolumeMountPath))
+		commandArgs = append(commandArgs, fmt.Sprintf("--etcd-client-key-path=%s/tls.key", common.EtcdClientTLSVolumeMountPath))
+	}
+	return commandArgs
+}
+
+func (b *stsBuilder) getEtcdContainerEnvVars() []corev1.EnvVar {
+	if b.useEtcdWrapper {
+		return []corev1.EnvVar{}
+	}
+	backTLSEnabled := b.etcd.Spec.Backup.TLS != nil
+	scheme := utils.IfConditionOr[string](backTLSEnabled, "https", "http")
+	endpoint := fmt.Sprintf("%s://%s-local:%d", scheme, b.etcd.Name, b.backupPort)
+
+	return []corev1.EnvVar{
+		{Name: "ENABLE_TLS", Value: strconv.FormatBool(backTLSEnabled)},
+		{Name: "BACKUP_ENDPOINT", Value: endpoint},
+	}
+}
+
+func (b *stsBuilder) getPodSecurityContext() *corev1.PodSecurityContext {
+	if !b.useEtcdWrapper {
+		return nil
+	}
+	return &corev1.PodSecurityContext{
+		RunAsGroup:   pointer.Int64(65532),
+		RunAsNonRoot: pointer.Bool(true),
+		RunAsUser:    pointer.Int64(65532),
+		FSGroup:      pointer.Int64(65532),
+	}
+}
+
+func (b *stsBuilder) getEtcdContainerSecretVolumeMounts() []corev1.VolumeMount {
+	secretVolumeMounts := make([]corev1.VolumeMount, 0, 6)
+	if b.etcd.Spec.Etcd.ClientUrlTLS != nil {
+		secretVolumeMounts = append(secretVolumeMounts,
+			corev1.VolumeMount{
+				Name:      common.EtcdCAVolumeName,
+				MountPath: common.EtcdCAVolumeMountPath,
+			},
+			corev1.VolumeMount{
+				Name:      common.EtcdServerTLSVolumeName,
+				MountPath: common.EtcdServerTLSVolumeMountPath,
+			},
+			corev1.VolumeMount{
+				Name:      common.EtcdClientTLSVolumeName,
+				MountPath: common.EtcdClientTLSVolumeMountPath,
+			},
+		)
+	}
+	if b.etcd.Spec.Etcd.PeerUrlTLS != nil {
+		secretVolumeMounts = append(secretVolumeMounts,
+			corev1.VolumeMount{
+				Name:      common.EtcdPeerCAVolumeName,
+				MountPath: common.EtcdPeerCAVolumeMountPath,
+			},
+			corev1.VolumeMount{
+				Name:      common.EtcdPeerServerTLSVolumeName,
+				MountPath: common.EtcdPeerServerTLSVolumeMountPath,
+			},
+		)
+	}
+	if b.etcd.Spec.Backup.TLS != nil {
+		secretVolumeMounts = append(secretVolumeMounts,
+			corev1.VolumeMount{
+				Name:      common.BackupRestoreCAVolumeName,
+				MountPath: common.BackupRestoreCAVolumeMountPath,
+			},
+		)
+	}
+	return secretVolumeMounts
+}
+
+// getPodVolumes gets volumes that needs to be mounted onto the etcd StatefulSet pods
+func (b *stsBuilder) getPodVolumes(ctx component.OperatorContext) ([]corev1.Volume, error) {
+	volumes := []corev1.Volume{
+		{
+			Name: common.EtcdConfigVolumeName,
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: b.etcd.GetConfigMapName(),
+					},
+					Items: []corev1.KeyToPath{
+						{
+							Key:  etcdConfigFileName,
+							Path: etcdConfigFileName,
```

Comment:
`Key` is the key of the specific item in the configmap, while `Path` is the path to store that item of the configmap in. Here, we use `etcd.conf.yaml` for both the key as well as the path, since we want the file `etcd.conf.yaml` to be created from the configmap item with key `etcd.conf.yaml` itself. If it makes it easier to read, I can add a simple comment stating this.